### PR TITLE
Clean up the support for nfs additional repositories

### PR DESCRIPTION
**Pull request by poncovka**
**IMPORTANT:** This pull request doesn't change behaviour of nfs repositories. It cleans up the current implementation and prepares the code for possible changes in the supported nfs formats.

* Reorganize the `parse_nfs_url` function.
* Handle invalid protocols during the payload setup.
* Don't require a specific nfs format during the payload setup.

**Related to:** https://github.com/rhinstaller/anaconda/discussions/4491

### CODE REVIEW

File: pyanaconda/payload/dnf/payload.py [hunk 1]
```diff
@@ -365,11 +365,15 @@ def _set_up_additional_repository(self, data):
         """Set up sources for the additional repository."""
         # Check the validity of the repository.
         if not data.url:
-            msg = "The '{}' repository has no mirror, baseurl or metalink set."
+            msg = _("The '{}' repository has no mirror, baseurl or metalink set.")
             raise SourceSetupError(msg.format(data.name)) from None
```
Comment (rvykydal) [inline pyanaconda/payload/dnf/payload.py:369]: Shouldn't we use hint for translators here? Something like:
```
            msg = _("The '{repository_name}' repository has no mirror, baseurl or metalink set.")
            raise SourceSetupError(msg.format(repository_name=data.name)) from None
```

File: pyanaconda/payload/dnf/payload.py [hunk 2]
```diff
@@ -405,6 +410,11 @@ def _set_up_additional_repository(self, data):
             )
             result = task.run()
             data.url = "file://" + result.install_tree_path
+            return
+
+        # Otherwise, raise an error.
+        msg = _("The '{}' repository has an unsupported protocol.")
+        raise SourceSetupError(msg.format(data.name)) from None
```
Comment (rvykydal) [inline pyanaconda/payload/dnf/payload.py:417]: The hint for translators would go also here (as above).

**Review by rvykydal — APPROVED** [review]
Maybe the hints for translators are worth considering,
otherwise looks good to me.

File: pyanaconda/payload/dnf/payload.py (hunk 2 above)
Comment (poncovka) [reply pyanaconda/payload/dnf/payload.py]: Fixed.

**Pull request merged**